src/traces/funnelarea/plot.js

```diff
@@ -94,36 +96,41 @@ module.exports = function plot(gd, cdModule) {
                         s.attr('data-notex', 1);
                     });
 
+                    var font = Lib.extendFlat({}, determineInsideTextFont(trace, pt, fullLayout.font), {});
+                    font.size = Math.max(font.size, fullLayout.uniformtext.minsize || 0);
+
                     sliceText.text(pt.text)
                         .attr({
                             'class': 'slicetext',
                             transform: '',
                             'text-anchor': 'middle'
                         })
-                        .call(Drawing.font, determineInsideTextFont(trace, pt, gd._fullLayout.font))
+                        .call(Drawing.font, font)
                         .call(svgTextUtils.convertToTspans, gd);
 
                     // position the text relative to the slice
                     var textBB = Drawing.bBox(sliceText.node());
                     var transform;
 
                     var x0, x1;
-                    var y0 = Math.min(pt.BL[1], pt.BR[1]);
-                    var y1 = Math.max(pt.TL[1], pt.TR[1]);
+                    var y0 = Math.min(pt.BL[1], pt.BR[1]) + cy;
+                    var y1 = Math.max(pt.TL[1], pt.TR[1]) + cy;
 
-                    x0 = Math.max(pt.TL[0], pt.BL[0]);
-                    x1 = Math.min(pt.TR[0], pt.BR[0]);
+                    x0 = Math.max(pt.TL[0], pt.BL[0]) + cx;
+                    x1 = Math.min(pt.TR[0], pt.BR[0]) + cx;
```

Comment:
These changes are pretty subtle. Can you explain why you needed to that?

Reply:
The text elements here used to have two `translate` calls. We need to reduce that to only one for keeping track of actual translation (in the transform object).

Reply:
Brilliant - thanks!